File: 0134-gas-station/0134-gas-station.cpp
class Solution {
public:
    int canCompleteCircuit(vector<int>& gas, vector<int>& cost) {
        int n = gas.size();
        
        int tot_gas = 0, tot_cost = 0;
        int curr_gas = 0, start = 0;;
        
        for(int i = 0; i < n; ++i) {
            tot_gas += gas[i];
            tot_cost += cost[i];
            
            curr_gas += (gas[i] - cost[i]);
            if(curr_gas < 0) {
                start = i+1;
                curr_gas = 0;
            }
        }
        return tot_gas < tot_cost ? -1 : start;
    }
};

File: 0134-gas-station/0134-gas-station.go
func canCompleteCircuit(gas []int, cost []int) int {
    // var n := len(gas)
    var tot_gas = 0
    var tot_cost = 0
    var curr_gas = 0
    var start = 0

    for i, _ := range gas {
        tot_gas += gas[i]
        tot_cost += cost[i]

        curr_gas += (gas[i] - cost[i])
        if(curr_gas < 0) {
            start = i+1
            curr_gas = 0
        }
    }
    if tot_gas < tot_cost {
        return -1 
    }
    return start
}